Jetchat/app/src/main/java/com/example/compose/jetchat/profile/ProfileViewModel.kt

```diff
@@ -35,8 +35,8 @@ class ProfileViewModel : ViewModel() {
         _userData.value = if (userId == meProfile.userId) meProfile else colleagueProfile
     }
 
-    private val _userData = MutableLiveData<ProfileScreenState>()
-    val userData: LiveData<ProfileScreenState> = _userData
+    private val _userData = MutableStateFlow<ProfileScreenState?>(null)
```

Comment:
This pattern is very common when migrating from LD to StateFlow but I'd discourage having a StateFlow of a nullable type. What about creating an empty ProfileScreenState for this?

```suggestion
    private val _userData = MutableStateFlow<ProfileScreenState>(
        ProfileScreenState("", "", "", "", "") // Empty user to begin with
    )
```

The view then should check `userId.isEmpty()` for showing the error state, or even better, have that function as an extension function on `ProfileScreenState` to indicate if it's the object is properly constructed or not.

The benefits of doing this might not be obvious for this example, but it's the right pattern to use. In this way, you avoid that you can emit `null` again for this `StateFlow` and handling nullability when collecting

Reply:
I think we're trying to avoid null too much, this is an acceptable case for it. An empty screen state doesn't mean the same thing as null. In the view I can do:
```
                        if (userData == null) {
                            ProfileError()
```
and display better data to the user than an empty profile. Or I would have to create an extension `ProfileScreenState.isEmpty()` or something similar which is wasteful.

In a real app you would probably have this data wrapped into a resource sealed class. If there's a big problem with nullable types I would do that first.

Reply:
ping!

Reply:
Fine, leave it like this 🤷‍♂️